packages/core/types/src/modules/entity-service/params/sort.ts

```diff
@@ -75,16 +75,24 @@ export type ArrayNotation<TSchemaUID extends Common.UID.Schema> = Any<TSchemaUID
  * type F = 'title'; // ❌
  */
 export type ObjectNotation<TSchemaUID extends Common.UID.Schema> = {
-  // First level sort
-  [key in SingleAttribute<TSchemaUID>]?: OrderKind.Any;
-} & {
-  // Deep sort, only add populatable keys that have a
-  // target (remove dynamic zones and other polymorphic links)
-  [key in Attribute.GetKeysWithTarget<TSchemaUID>]?: ObjectNotation<
-    Attribute.GetTarget<TSchemaUID, key>
-  >;
+  [TKey in ObjectNotationKeys<TSchemaUID>]?: TKey extends SingleAttribute<TSchemaUID>
+    ? // First level sort (scalar attributes, id, ...)
+      OrderKind.Any
+    : // Deep sort (relations with a target, components, media, ...)
+      ObjectNotation<Attribute.GetTarget<TSchemaUID, TKey>>;
```

Comment:
Considering the key type is not explicit but coming from another type  ObjectNotationKeys, it could get changed and the conditions below would not work anymore. 

I would add a TKey extends Attribute.GetKeysWithTarget<TSchemaUID> to be exhaustive and avoid breaking if someday we change ObjectNotationKeys

Reply:
Yeah, I thought about that and told myself it's unlikely etc... But you're right, it's probably better to be safe. I'll add it

Reply:
done in [64d75d0](https://github.com/strapi/strapi/pull/19093/commits/64d75d007fb872c50f0a37eed1518092ba61f425)